flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -752,6 +757,38 @@ public StatementSet createStatementSet() {
         return new StatementSetImpl(this);
     }
 
+    @Override
+    public CompiledPlan loadPlan(PlanReference planReference) throws IOException {
+        return planner.load(planReference);
+    }
+
+    @Override
+    public CompiledPlan compilePlanSql(String stmt) {
+        List<Operation> operations = getParser().parse(stmt);
+
+        if (operations.size() != 1 || !(operations.get(0) instanceof ModifyOperation)) {
+            throw new TableException(UNSUPPORTED_QUERY_IN_COMPILE_PLAN_SQL_MSG);
+        }
+
+        return planner.compile(Collections.singletonList((ModifyOperation) operations.get(0)));
+    }
+
+    @Override
+    public TableResult executePlan(CompiledPlan plan) {
+        List<Transformation<?>> transformations = planner.translate(plan);
+        List<String> sinkIdentifierNames = new ArrayList<>();
+        for (int i = 0; i < transformations.size(); ++i) {
+            // TODO serialize the sink table names to json plan ?
```

Comment:
I think the names are already part of the JSON plan. The question is rather whether we can somehow get the names from the transformations. We should definitely fix this TODO. Because otherwise the job name might not be deterministic.

Reply:
Can you check it now?